llvm/include/llvm/MC/MCRegisterInfo.h

```diff
@@ -353,6 +355,12 @@ class LLVM_ABI MCRegisterInfo {
     L2CVRegs[LLVMReg] = CVReg;
   }
 
+  void ignoreCVReg(MCRegister LLVMReg) { IgnoredCVRegs.insert(LLVMReg); }
+
+  bool isIgnoredCVReg(MCRegister LLVMReg) const {
+    return IgnoredCVRegs.contains(LLVMReg);
+  }
```

Comment:
I think you could skip the `DenseSet` and just do: 
```suggestion
  bool isIgnoredCVReg(MCRegister LLVMReg) const {
    return (LLVMReg >=  AArch64::Z0 && LLVMReg <= AArch64::Z31)
                || (LLVMReg >= AArch64::P0 && LLVMReg <= AArch64::P15);
  }
```

Reply:
Pretty sure that breaks layering; the AArch64 register names are only defined in the AArch64 backend.  I could make it a virtual method, I guess?

Reply:
Ah, I didn't spot this was outside the AArch64 backend, but yes, you could make it virtual or allow registering a `std::function/llvm::function_ref` callback to filter registers.

Reply:
Changed to virtual method.